src/Components/Locations.js

```diff
@@ -0,0 +1,55 @@
+import axios from 'axios';
+import './Locations.css'
+import { Component } from 'react';
+
+class Locations extends Component {
+    state = {
+        locations: [],
+        showLocations: false
+    }
+
+    getLocations = async () => {
+        try {
+            const { data } = await axios.get(
+                'https://ghibliapi.herokuapp.com/locations'
+            )
+            console.log(data)
+            this.setState({
+                locations: data,
+                showLocations: !this.state.showLocations
+            })
+
+        }
+        catch (error) {
+            this.setState({
+                locations: [],
+                showLocations: false
+            })
+        }
+    }
+
+
+    render() {
+        const { locations, showLocations } = this.state
+        const locationList = locations.map(location => {
+            return (
+                <li key={location.id}>
+                    Name: {location.name} <br />
+                    Climate: {location.climate}<br />
+                    Terrain: {location.terrain}
+                </li>
+            )
+        })
+        return (
+            <div>
+                <h1>List of Locations</h1>
+                <button onClick={this.getLocations}>{showLocations ? 'Hide Locations' : 'Show Locations'}</button>
```

Comment:
It seems like overkill to fetch the list of locations every time you click the button, even if you just want to hide the location list. Could you write a separate click handler that toggles `showLocations` without fetching new data?